.travis.yml

```diff
@@ -1,47 +1,45 @@
-language: android
-jdk: oraclejdk8
-
-# Choosing faster environment since it's a full VM.
-# https://docs.travis-ci.com/user/reference/overview/
 sudo: required
-dist: trusty
+language: generic
 
 env:
+  global:
+  - ABI="google_apis;armeabi-v7a"
+  - BUILD_TOOLS_VERSION=28.0.3
+  - ADB_INSTALL_TIMEOUT=25
+  - QEMU_AUDIO_DRV=none
   matrix:
-    # Notice matrix environment variables are not available during the install phase for android >
-    # components.
-    # These CANNOT BE USED in the android > components section, only in before_script and script.
-    - TEST_TYPE=unit
-    - TEST_TYPE=instrumentation EMULATOR_TAG=android EMULATOR_API=16 ABI=armeabi-v7a
-    - TEST_TYPE=instrumentation EMULATOR_TAG=android EMULATOR_API=19 ABI=armeabi-v7a
-    - TEST_TYPE=instrumentation EMULATOR_TAG=android EMULATOR_API=21 ABI=armeabi-v7a
-
-before_cache:
-  - rm -f  $HOME/.gradle/caches/modules-2/modules-2.lock
-  - rm -fr $HOME/.gradle/caches/*/plugin-resolution/
-
-cache:
-  directories:
-    - $HOME/.gradle/caches/
-    - $HOME/.gradle/wrapper/
-    - $HOME/.android/build-cache
-
-android:
-  components:
-    # https://github.com/travis-ci/travis-ci/issues/6040#issuecomment-219367943
-    - tools
-    - tools
-
-before_install:
+    - EMULATOR_API=16
+    - EMULATOR_API=19
+    - EMULATOR_API=21
+    - EMULATOR_API=24
+
+install:
+  - export ANDROID_HOME=~/android-sdk-linux
+  - wget -q "https://dl.google.com/android/repository/sdk-tools-linux-4333796.zip" -O android-sdk-tools.zip
+  - unzip -q android-sdk-tools.zip -d ${ANDROID_HOME}
+  - rm android-sdk-tools.zip
+  - PATH=${PATH}:${ANDROID_HOME}/tools:${ANDROID_HOME}/tools/bin:${ANDROID_HOME}/platform-tools
+  - yes | sdkmanager --update
+  - yes | sdkmanager --licenses
+  - ./gradlew dependencies || true
+  - sdkmanager "emulator" "tools" "platform-tools" > /dev/null
+  - sdkmanager --list | head -20
+  - sdkmanager "build-tools;$BUILD_TOOLS_VERSION" "platforms;android-$28" > /dev/null
+  - sdkmanager "system-images;android-$EMULATOR_API;$ABI" > /dev/null
+  - sdkmanager --list | head -15
   - mkdir "$ANDROID_HOME/licenses" || true
-  - echo -e "\n8933bad161af4178b1185d1a37fbf41ea5269c55\nd56f5187479451eabf01fb78af6dfcb131a6481e" > "$ANDROID_HOME/licenses/android-sdk-license"
-  - echo -e "\n84831b9409646a918e30573bab4c9c91346d8abd\n504667f4c0de7af1a06de9f4b1727b84351f2910" > "$ANDROID_HOME/licenses/android-sdk-preview-license"
+  - echo -e "\n8933bad161af4178b1185d1a37fbf41ea5269c55" > "$ANDROID_HOME/licenses/android-sdk-license"
+  - echo -e "\n84831b9409646a918e30573bab4c9c91346d8abd" > "$ANDROID_HOME/licenses/android-sdk-preview-license"
 
 before_script:
-  - ./travis_before_script.sh
+  - echo no | avdmanager create avd --force -n test -k "system-images;android-$EMULATOR_API;$ABI"
+  - ./gradlew assembleDebugAndroidTest -PdisablePreDex --continue --stacktrace
+  - $ANDROID_HOME/emulator/emulator -avd test -no-window &
+  - pwd
```

Comment:
is this and ls leftover from testing?

Reply:
Yes, thanks for spotting it for me

Reply:
Now the unnecessary commands are removed...